File: storage_sdk/src/main/java/com/walker/storage/room/repository/WordRepository.java
package com.walker.storage.room.repository;

import android.content.Context;

import androidx.lifecycle.LiveData;

import com.walker.storage.room.WalkerRoomDatabase;
import com.walker.storage.room.dao.WordDao;
import com.walker.storage.room.model.Word;

import java.util.List;

/**
 * Created by walkerzpli on 2021/8/5.
 */
public class WordRepository {

    private WordDao mWordDao;
    private LiveData<List<Word>> mAllWords;

    public WordRepository(Context context) {
        WalkerRoomDatabase db = WalkerRoomDatabase.getDatabase(context);
        mWordDao = db.wordDao();
        mAllWords = mWordDao.getAlphabetizedWords();
    }

    public LiveData<Word> getLastWord() {
        return mWordDao.getLastWord();
    }

    public LiveData<List<Word>> getAllWords() {
        return mAllWords;
    }

    public void insert(Word word) {
        WalkerRoomDatabase.databaseWriteExecutor.execute(() -> mWordDao.insert(word));
    }

    public void delete(String content) {
        WalkerRoomDatabase.databaseWriteExecutor.execute(() -> mWordDao.delete(content));
    }
}

File: storage-ktx/src/main/java/com/demo/storage_ktx/repository/WordRepository.kt
package com.demo.storage_ktx.repository

import androidx.annotation.WorkerThread
import com.demo.storage_ktx.model.Word
import com.demo.storage_ktx.dao.WordDao
import kotlinx.coroutines.flow.Flow

/**
 * Created by lizhiping on 2023/6/5.
 * <p>
 * description
 */
class WordRepository(private val wordDao: WordDao) {

    val allWords: Flow<List<Word>> = wordDao.getAlphabetizedWords()

    @Suppress("RedundantSuspendModifier")
    @WorkerThread
    suspend fun insert(word: Word) {
        wordDao.insert(word)
    }

    companion object {
        private const val TAG = "WordRepository"
    }
}